src/shared/loop-util.c

```diff
@@ -133,7 +133,19 @@ static int loop_configure_verify(int fd, const struct loop_config *c) {
                         return -errno;
 
                 if (z != c->info.lo_sizelimit) {
-                        log_debug("LOOP_CONFIGURE is broken, doesn't honour .lo_sizelimit. Falling back to LOOP_SET_STATUS64.");
+                        log_debug("LOOP_CONFIGURE is broken, doesn't honour .info.lo_sizelimit. Falling back to LOOP_SET_STATUS64.");
+                        broken = true;
+                }
+        }
+
+        if (c->block_size != 0) {
+                uint32_t z;
+
+                if (ioctl(fd, BLKSSZGET, &z) < 0)
```

Comment:
hmm, the kernel writes an `int` if the BLKSSZGET ioctl is called. We probably should use the same type here, even if it makes little sense to have signed block sizes...